src/share/parallel.c

```diff
@@ -49,9 +49,17 @@ compress_init_par(zfp_stream* stream, const zfp_field* field, uint chunks, uint
 
   /* set up buffer for each thread to compress to */
   bs = (bitstream**)malloc(chunks * sizeof(bitstream*));
+  if (!bs) {
+    fprintf(stderr, "cannot allocate memory for per-thread bit streams\n");
+    exit(EXIT_FAILURE);
```

Comment:
I'm afraid that exiting the entire application on failure is not acceptable.  zfp also should not write to stderr.  We need to return NULL here and have the caller check the return value.